app/controllers/api/tags_controller.rb

```diff
@@ -0,0 +1,87 @@
+module Api
+  # Api controller for tags
+  # Uses Rails' RESTful routes (check 'rake routes' for the configured routes)
+  class TagsController < MainApiController
+    DEFAULT_FIELDS = [:id, :name, :description, :creator, :use].freeze
+
+    def index
+      @assignment = Assignment.find_by(id: params[:assignment_id])
+
+      respond_to do |format|
+        parent = @assignment || current_course
+        tags = parent.tags.includes(:role, :groupings).order(:name)
+
+        tag_info = tags.map do |tag|
+          {
+            id: tag.id,
+            name: tag.name,
+            description: tag.description,
+            creator: tag.role.display_name,
+            use: tag.groupings.size
+          }
+        end
+        format.xml do
+          render xml: tag_info
+        end
+        format.json do
+          render json: tag_info
+        end
+      end
+    end
+
+    # Creates a new instance of the tag.
+    def create
+      params.permit(:name, :description)
+      begin
+        assignment = current_course.assignments.find_by(id: params[:assignment_id])
+        new_tag = Tag.new(course: current_course, role: current_role, assessment: assignment, **self.tag_params)
+        if new_tag.save && params[:grouping_id]
+          grouping = assignment.groupings.find(params[:grouping_id])
+          grouping.tags << new_tag
+        end
+      rescue StandardError => e
+        render 'shared/http_status', locals: { code: '422', message: e.to_s }, status: :unprocessable_entity
+      else
+        render 'shared/http_status',
+               locals: { code: '201', message: HttpStatusHelper::ERROR_CODE['message']['201'] }, status: :created
+      end
+    end
+
+    def update
+      tag = Tag.find(params[:id])
```

Comment:
use `find_by(id: ...)` instead of `find`; the latter raises an error, while the former will return `nil`, which is what you want here